src/simulation/thermal/sensorThermal/_UnitTest/test_sensorThermal.py

```diff
@@ -0,0 +1,229 @@
+
+    In this file we are checking the values of the variables
+
+    - ``sensorTemp``
+
+    which represents the temperature of the sensor. This array is compared to the ``truthTemp`` array, which contains
+    the true values of the temperature.
+    """
+    [testResults, testMessage] = sensorThermalTest(show_plots, accuracy)
+    assert testResults < 1, testMessage
+
+
+def sensorThermalTest(show_plots, accuracy):
+    testFailCount = 0  # zero unit test result counter
+    testMessages = []  # create empty list to store test log messages
+
+    # Create simulation variable names
+    unitTaskName = "unitTask"  # arbitrary name (don't change)
+    unitProcessName = "TestProcess"  # arbitrary name (don't change)
+
+    #   Create a sim module as an empty container
+    unitTestSim = SimulationBaseClass.SimBaseClass()
+
+    # set the simulation time variable used later on
+    simulationTime = macros.sec2nano(1.)
+
+    #
+    #  create the simulation process
+    #
+
+    testProcessRate = macros.sec2nano(1.0)  # update process rate update time
+    testProc = unitTestSim.CreateNewProcess(unitProcessName)
+    testProc.addTask(unitTestSim.CreateNewTask(unitTaskName, testProcessRate))
+
+    #
+    #   setup the simulation tasks/objects
+    #
+
+    #  set device status message
+    sensorStatusMsgPayload = messaging.DeviceStatusMsgPayload()
+    sensorStatusMsgPayload.deviceStatus = 1
+    sensorStatusMsg = messaging.DeviceStatusMsg().write(sensorStatusMsgPayload)
+
+    #  set the spacecraft message
+    scStateMsgPayload = messaging.SCStatesMsgPayload()
+    scStateMsgPayload.r_BN_N = [6378*1000., 0., 0.]
+    scStateMsgPayload.sigma_BN = [0., 0., 0.]
+    scStateMsg = messaging.SCStatesMsg().write(scStateMsgPayload)
+
+    #  set the sun message
+    sunMsgPayload = messaging.SpicePlanetStateMsgPayload()
+    sunMsgPayload.PositionVector = [astroFunctions.AU*1000., 0., 0.]
+    sunMsg = messaging.SpicePlanetStateMsg().write(sunMsgPayload)
+
+
+    #
+    #   Setup the temperature modeling
+    #
+    sensorThermalModel = sensorThermal.SensorThermal()
+    sensorThermalModel.ModelTag = 'sensorThermalModel'
+    nHat_B = [0, 0, 1]
+    sensorArea = 1.0  # m^2
+    sensorAbsorptivity = 0.25
+    sensorEmissivity = 0.34
+    sensorMass = 2.0  # kg
+    sensorSpecificHeat = 890
+    sensorPowerDraw = 30.0  # W
+    sensorThermalModel.T_0 = 0  # [ºC]
+    sensorThermalModel.setSensorParameters(nHat_B, sensorArea, sensorAbsorptivity, sensorEmissivity, sensorMass, sensorSpecificHeat)
+    sensorThermalModel.sensorPowerDraw = sensorPowerDraw
+    sensorThermalModel.sunInMsg.subscribeTo(sunMsg)
+    sensorThermalModel.stateInMsg.subscribeTo(scStateMsg)
+    sensorThermalModel.sensorStatusInMsg.subscribeTo(sensorStatusMsg)
+    unitTestSim.AddModelToTask(unitTaskName, sensorThermalModel)
+
+    #
+    # log data
+    #
+
+    # log the RW temperature
+    tempLog = sensorThermalModel.temperatureOutMsg.recorder()
```

Comment:
:large_blue_circle:  As discussed, this is three tests hiding in one. No action needed from you here. I'll look into ways to enhance our appropriate use of pytests fixtures etc. A good article about good unit test https://levelup.gitconnected.com/how-to-write-clear-and-robust-unit-tests-the-dos-and-donts-5021c097d041

Reply:
I'm not following that comment.  He is changing the inputs over a few time steps and checking that the module output is correct.  I'm fine with this test as constructed.

Reply:
There's no change needed here. My question was that the `numTests` variable is incremented, where as pytest only sees a single test. I know and understand the reason for it. I just see it as an end run around the pytest framework. As I mentioned to @ai-in-aerospace I will ponder this some more myself to see how we can support all current uses cases Basilisk has for tests and reporting, and increase our use of the wonderful features pytest offers.

Reply:
Ah, I see what you were looking at.  In this code `numTests` is not really defining 3 separate tests in one.  The module is reading in different input states and then integrating.  Thus, the module is making sure we get the expected values over 3 separate time steps.  We can't test the functionality of this module over a single time step.  Thus, to me this is really one single test of the module functionality.